enhancements/network/ovn-observability-api.md

```diff
@@ -0,0 +1,290 @@
+  can't be used, as observability module doesn't watch any objects.
+
+#### ObservabilityConfig CRD
+
+The CRD for this feature will be a part of ovn-kubernetes to allow using it upstream.
+
+```go
+
+// ObservabilityConfig described OVN Observability configuration.
+//
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=observabilityconfig,scope=cluster
+// +kubebuilder:singular=observabilityconfig
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type ObservabilityConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ObservabilitySpec `json:"spec"`
+	// +optional
+	Status ObservabilityStatus `json:"status,omitempty"`
+}
+
+type ObservabilitySpec struct {
+	// CollectorID is the ID of the collector that should be unique across the cluster, and will be used to receive configured samples.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=1
+	// +required
+	CollectorID int32 `json:"collectorID"`
+	// Features is a list of Observability features that can generate samples and their probabilities for a given collector.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +required
+	Features []FeatureConfig `json:"features"`
+
+	// Filter allows to apply ObservabilityConfig in a granular manner.
+	// +optional
+	Filter *Filter `json:"filter,omitempty"`
+}
+
+type FeatureConfig struct {
+	// Probability is the probability of the feature being sampled in percent.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=100
+	// +required
+	Probability int32
+	// Feature is the Observability feature that should be sampled.
+	// +kubebuilder:validation:Required
+	// +required
+	Feature ObservabilityFeature
+}
+
+// ObservabilityStatus contains the observed status of the ObservabilityConfig.
+type ObservabilityStatus struct {
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// Filter allows to apply ObservabilityConfig in a granular manner.
+// Currently, it supports node and namespace based filtering.
+// If both node and namespace filters are specifies, they are logically ANDed.
+// +kubebuilder:validation:MinProperties=1
+type Filter struct {
+	// nodeSelector applies ObservabilityConfig only to nodes that match the selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+	// namespaces is a list of namespaces to which the ObservabilityConfig should be applied.
+	// It only applies to the namespaced features, currently that includes NetworkPolicy and EgressFirewall.
+	// +kubebuilder:MinItems=1
+	// +optional
+	Namespaces *[]string `json:"namespaces,omitempty"`
```

Comment:
maybe this is more of a future topic for discussion, but I was just thinking...wouldn't another good filter be on UDN? That way i know my users in network A are having problems i just want to turn on sampling for netpol in network A, which would then translate to on the backend to every network policy for namespaces where that network is active?

Reply:
Yeah I think it is technically doable, and also can be considered an extension for config- or feature-wide filter